Xamarin.Forms.Core/VisualElement.cs

```diff
@@ -181,16 +183,18 @@ void StopNotifyingBackgroundChanges()
 			if (Background != null)
 			{
 				Background.PropertyChanged -= OnBackgroundChanged;
-				Background.PropertyChanging -= OnBackgroundChanging;
+
+				if (Background is GradientBrush gradientBrush)
+					gradientBrush.InvalidateGradientBrushRequested -= InvalidateGradientBrushRequested;
```

Comment:
What about brushes that are used for the Stroke or Fill properties of a shape - do you need similar code there?

Reply:
Tested a Shape with Stroke and when i change it works fine.